halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/providers/ProviderCommand.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1.providers;
+
+import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.command.v1.NestableCommand;
+import com.netflix.spinnaker.halyard.cli.command.v1.providers.dockerRegistry.DockerRegistryCommand;
+import com.netflix.spinnaker.halyard.cli.command.v1.providers.kubernetes.KubernetesCommand;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is a top-level command for dealing with your halconfig.
+ *
+ * Usage is `$ hal config`
+ */
+@Parameters()
+public class ProviderCommand extends NestableCommand {
+  @Getter(AccessLevel.PROTECTED)
+  private Map<String, NestableCommand> subcommands = new HashMap<>();
+
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "provider";
+
+  public ProviderCommand() {
+    KubernetesCommand kubernetesCommand = new KubernetesCommand();
```

Comment:
The problem is that each provider will have their own set of flags that the maintainers see as valuable, each with their own update semantics. In the end, each provider will be forced to update accounts in the same way, namely:

1. Get the current account's definition
2. Make changes to that account's definitions given what flags were set
3. PUT the new definition to the daemon, and report if validation succeeds

The problem is, each provider will have different flags to set and in different ways. Auto generating these flags will have me guessing at the intentions of providers in ways I can't predict.

For example, I can pass --context to the kubernetes provider, but when I do that, I shouldn't specify --user. This is unique to kubernetes. In the docker provider to have a reasonable user flow, you should enforce that username & password are changed in tandem. AWS regions should be smart about which aws zones they prefer vs. which are required, etc...